packages/infra/src/constructs/stack.ts

```diff
@@ -35,6 +35,35 @@ export class HookStack extends Stack {
           'codedeploy:PutLifecycleEventHookExecutionStatus',
           'lambda:InvokeFunction',
         ],
+        effect: aws_iam.Effect.ALLOW,
+        resources: ['*'],
+      }),
+    );
+
+    // Deny access to resources that lack an `aws-codedeploy-hooks` tag.
+    beforeAllowTrafficHook.addToRolePolicy(
+      new aws_iam.PolicyStatement({
+        actions: ['*'],
+        conditions: {
+          Null: {
+            'aws:ResourceTag/aws-codedeploy-hooks': 'true',
+          },
+        },
+        effect: aws_iam.Effect.DENY,
+        resources: ['*'],
+      }),
+    );
+
+    // Deny access to resources that have a falsy `aws-codedeploy-hooks` tag.
+    beforeAllowTrafficHook.addToRolePolicy(
+      new aws_iam.PolicyStatement({
+        actions: ['*'],
+        conditions: {
+          StringEquals: {
```

Comment:
Should we consider StringNotEquals true?

Reply:
I was thinking of sneaking in the package version as the tag value, i.e. `aws-codedeploy-hooks: 1.2.3` or perhaps even `aws-codedeploy-hooks: 1.2.3-abcdefg`

As a point of comparison, Gantry writes numerous tags like

```
gantry:resource-kind: environment
gantry:tool-commit: abcdefg
gantry:tool-version: v1.2.3
```

Reply:
Ah makes sense :+1: